packages/eas-cli/src/build/build.ts

```diff
@@ -274,7 +282,28 @@ async function uploadProjectAsync<TPlatform extends Platform>(
             completedMessage: (duration: string) => `Uploaded to EAS ${chalk.dim(duration)}`,
           })
         );
-        return bucketKey;
+        try {
+          const projectMetadataFile = await makeProjectMetadataFileAsync(projectTarball.path);
+
+          const metadataLocation = await uploadFileAtPathToGCSAsync(
+            ctx.graphqlClient,
+            UploadSessionType.EasBuildGcsProjectSources,
+            projectMetadataFile.path,
+            createProgressTracker({
+              total: projectMetadataFile.size,
+              message: ratio =>
+                `Uploading metadata to EAS Build (${formatBytes(
+                  projectMetadataFile.size * ratio
+                )} / ${formatBytes(projectMetadataFile.size)})`,
+              completedMessage: (duration: string) => `Uploaded to EAS ${chalk.dim(duration)}`,
+            })
+          );
+          return { bucketKey, metadataLocation };
+        } catch (e) {
+          Log.warn('Failed to upload metadata to EAS Build');
+          Log.debug(e);
+          return { bucketKey };
+        }
```

Comment:
Maybe we want also to delete the metadata file the same way we do here https://github.com/expo/eas-cli/pull/2149/files#diff-a19f6ae31352620fe175af46b2a62d16c12d93f2b1014d59ccbf97fe45ac08baR322-R324